src/MessageContainer.js

```diff
@@ -95,34 +51,36 @@ export default class MessageContainer extends React.Component {
     return null;
   }
 
-  renderFooter() {
-    if (this.props.renderFooter) {
-      const footerProps = {
-        ...this.props,
-      };
-      return this.props.renderFooter(footerProps);
+  scrollTo(options) {
+    if (this.flatListRef) {
+      this.flatListRef.scrollToOffset(options);
     }
-    return null;
   }
 
-  renderRow(message) {
-    if (!message._id && message._id !== 0) {
-      console.warn('GiftedChat: `_id` is missing for message', JSON.stringify(message));
+  renderRow({ item, index }) {
+    if (!item._id && item._id !== 0) {
+      console.warn('GiftedChat: `_id` is missing for message', JSON.stringify(item));
     }
-    if (!message.user) {
-      if (!message.system) {
-        console.warn('GiftedChat: `user` is missing for message', JSON.stringify(message));
+    if (!item.user) {
+      if (!item.system) {
+        console.warn('GiftedChat: `user` is missing for message', JSON.stringify(item));
       }
-      message.user = {};
+      item.user = {};
     }
+    const { messages, ...restProps } = this.props;
+    const previousMessage = messages[index + 1] || {};
+    const nextMessage = messages[index - 1] || {};
 
     const messageProps = {
-      ...this.props,
-      key: message._id,
-      currentMessage: message,
-      previousMessage: message.previousMessage,
-      nextMessage: message.nextMessage,
-      position: message.user._id === this.props.user._id ? 'right' : 'left',
+      ...restProps,
+      key: item._id,
+      currentMessage: item,
+      previousMessage,
+      nextMessage,
+      position: item.user._id === this.props.user._id ? 'right' : 'left',
+      hash: sum(
```

Comment:
What is the purpose of `hash`? As far as I'm aware, the `Message` component does not expect a `hash` prop.